components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiMemoryUsageMetric.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.carbon.siddhi.metrics.core;
+
+import org.wso2.carbon.metrics.core.Gauge;
+import org.wso2.carbon.metrics.core.Level;
+import org.wso2.carbon.metrics.core.MetricService;
+import org.wso2.carbon.siddhi.metrics.core.internal.SiddhiMetricsDataHolder;
+import org.wso2.carbon.siddhi.metrics.core.internal.SiddhiMetricsManagement;
+import org.wso2.siddhi.core.util.statistics.MemoryUsageTracker;
+import org.wso2.siddhi.core.util.statistics.memory.ObjectSizeCalculator;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static org.wso2.carbon.metrics.core.Level.INFO;
+import static org.wso2.carbon.metrics.core.Level.OFF;
+
+/**
+ * Siddhi Memory usage MMetrics Tracker.
+ */
+public class SiddhiMemoryUsageMetric implements MemoryUsageTracker {
+    private static final String METRIC_SUFFIX_MEMORY = "memory";
+    private ConcurrentMap<Object, ObjectMetric> registeredObjects = new ConcurrentHashMap<Object, ObjectMetric>();
+    private MetricService metricService;
+    private String siddhiAppName;
+    private boolean statisticEnabled;
+
+    public SiddhiMemoryUsageMetric(MetricService metricService, String siddhiAppName,boolean isStatisticEnabled) {
```

Comment:
Please add a space next to the comma

Reply:
Fixed in 557a2749c36e79d0b3f9922b371aca95c427766f